### ISSUE TITLE
Require not working as expected

### ISSUE THREAD

**Issue by tax** [chilts/awssum#1]
First of all thank you for creating this wonderful library. 

After installing with npm I was not able to import the modules as I expected and looking at the examples:

``` javascript
//Does not work taken from example 
var amazon = require("amazon/amazon");
var sqs = require("amazon/sqs");

//Also does not work
var amazon = require("awssum/amazon/amazon");
var sqs = require("awssum/amazon/sqs");
```

This however works

``` javascript
var amazon = require("awssum/lib/amazon/amazon");
var sqs = require("awssum/lib/amazon/sqs");
```

Cheers

**Comment by chilts** [comment]
No worries and thanks for the report.

You're right, the way I'm making people require services is completely wrong. What I want to do for the next version is something like this:

```
var awssum = require('awssum');

var sqs = awssum.load('amazon', 'sqs');
var elb = awssum.load('amazon', 'elb');
var ec2 = awssum.load('amazon', 'ec2');
var rsAuthentication = awssum.load('rackspace', 'Authentication');
var rsFiles = awssum.load('rackspace', 'files');
var rsServers = awssum.load('rackspace', 'servers');
var gh = awssum.load('github', 'github');
```

...etc...

I hope that's ok and that you can live with the above for the moment. I'm not far from releasing the next version so I'll ping you on here once that's done.

Cheers. :)

**Comment by tax** [comment]
Thanks for the feedback I like your solution. I was wondering how this is done in other libs and they use index.js files to and export the module there:

https://github.com/LearnBoost/mongoose/blob/master/lib/index.js

Regards Paul

**Comment by chilts** [comment]
Ah, yeah I've seen ones like that before ... though I'm undecided as to what I do next. On the one hand I want people to be able to add new APIs to awssum (which counts out this idea) without them being inside the awssum repo (so my .load() idea works).

On the other, that is probably too hard to do and not very practical, which means your idea wins. Let me stew on it for a few days. I'm literally really close to v0.2.0 so I expect to have it out within a couple of weeks. Thinking about it now, I think your idea is best so I'll add it for the next release.

Many thanks for your help and ideas. :)

**Comment by chilts** [comment]
Hi Paul,

This is now fixed in the v0.3.x series. You can do the following:

var awssum = require('awssum');
var amazon = awssum.load('amazon/amazon');
var sdbService = awssum.load('amazon/simpledb');
// ...etc...

Cheers,
Andy

**Comment by tax** [comment]
Just updated to the new version thanks a lot Andy.
